### PR TITLE
[feat] Add multiplatform docker support

### PR DESCRIPTION
**Pull request by treff7es**
- Multiplatform capable docker build for datahub frontend, datahub gms, kafka setup
- Custom arm64 supported docker compose file for m1
- Add gradle parameter to use systemwide node and not install -> gradle build is unable to install node on arm64 in docker

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: datahub-web-react/build.gradle [hunk 1]
```diff
@@ -5,10 +5,15 @@ node {
...
+
+  if (project.hasProperty('useSystemNode') && project.getProperty('useSystemNode').toBoolean()) {
+    download = false
+  } else {
+    download = true
+  }
 
   // Version of node to use.
-  version = '14.15.3'
+  version = '16.8.0'
```
Comment (dexter-mh-lee) [inline datahub-web-react/build.gradle:16]: @gabe-lyons is this node upgrade ok?

File: docker/quickstart/docker-compose-without-neo4j-m1.quickstart.yml [hunk 2]
```diff
@@ -0,0 +1,161 @@
+networks:
```
Comment (dexter-mh-lee) [inline docker/quickstart/docker-compose-without-neo4j-m1.quickstart.yml:1]: Quickstart files should only be created through the python code that generates these files. Can you add logic to this file to add this m1 file? https://github.com/linkedin/datahub/blob/9cfc16bc85688b999a1fe98944ea8dedf5d66246/docker/quickstart/generate_docker_quickstart.py

File: datahub-web-react/build.gradle (hunk 1 above)
Comment (gabe-lyons) [reply datahub-web-react/build.gradle:16]: yes! I've been running datahub using 16.4.2 for a while and it's been just fine

File: docker/kafka-setup/Dockerfile [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # This "container" is a workaround to pre-create topics
-FROM confluentinc/cp-kafka:6.2.0
+FROM kymeric/cp-kafka:latest
```
[the remark this answers is not in the document]
Comment (shirshanka) [reply docker/kafka-setup/Dockerfile:2]: Yes-ish + we're only using this for the kafka-setup container, which we can slim down significantly after getting this PR merged.

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**